chapter05/chapter05.adoc

```diff
@@ -0,0 +1,596 @@
+
+- If the service is not in maintenance mode, then the method checks if the list of retrieved products is not `null` and not empty. 
+
+- If products are found, it constructs a Response with a status of `200 OK` and includes the list of products as the response entity. This indicates a successful operation where product data is found and returned. 
+
+- If the products list is`null` or empty, indicating no products were found, the method constructs and returns a `Response` with a `404 Not Found` status code and a message stating that no products were found.
+
+When we deploy the application and invoke the `/api/products` endpoint, we should see the list of products as below:
+
+[source, json]
+----
+[{"description":"Apple iPhone 15 Pro","id":1,"name":"iPhone 15 Pro","price":999.0}]
+----
+
+=== Specifying Default Values for a `ConfigProperty`
+
+For non-critical properties, providing a default value using the defaultValue attribute of the @ConfigProperty annotation ensures that your application has a fallback option. We can specify a default value to be used if the property does not exist as below:
+
+[source, java]
+----
+public class ProductResource {
+
+   @Inject
+   @ConfigProperty(name="product.maintenanceMode", defaultValue="false")
+   private boolean maintenanceMode;
+   … 
+----
+
+In the example above , the `false` default value will be used if the `product.maintenanceMode` property does not exist.
+
+=== Type Conversion in `ConfigProperty`
+
+`ConfigProperty` also supports type conversion, so we can inject our configuration data into fields of any type:
+
+[source, java]
+----
+   @Inject
+   @ConfigProperty(name="product.maintenanceMode", defaultValue="false")
+   private boolean maintenanceMode;
+----
+
+In this example, the `product.maintenanceMode` property will be converted to an `Boolean` before it is injected into the `maintenanceMode` field. If the conversion fails, a `org.eclipse.microprofile.config.ConversionFailedException` will be thrown.
+
+=== Converting Configuration data to a POJO
+
+We can also use the Config API to convert our configuration data to a POJO:
+
+[source, java]
+----
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+
+public class MyApplication {
+    @Inject
+    private MaintenanceMessage message;
+}
+
+
+public class MaintenanceMessage {
+    @ConfigProperty(name="product.maintenanceMessage")
+    private String message;
+}
+----
+
+In this example, we’re injecting a property named "product.maintenanceMessage" into the message field of our MaintenanceMessage class. If the property does not exist, a `org.eclipse.microprofile.config.ConfigPropertyNotFoundException` will be thrown.
```

Comment:
The exception does not exist.

Reply:
Removed.

Reply:
I still see this exception on the line 325.